### PR TITLE
Update filter job creation to match new filter api spec

### PR DESCRIPTION
**Pull request by mattrout92**
### What

Update filter job creation to match new filter api spec
### How to review

Create a filter job through the web UI
Add dimension options to the job
Submit the job and verify the url changes to /filter-ouputs/{id}
Verify that when you click the adjust filter options button you return to /filters/{id}/dimensions

### Who can review

Anyone **DO NOT MERGE UNTIL BACKEND SERVICES HAVE BEEN MERGED**

### CODE REVIEW

File: handlers/handlers.go [hunk 1]
```diff
@@ -91,13 +85,15 @@ func CreateFilterID(c FilterClient, dc DatasetClient) http.HandlerFunc {
...
-				if err = c.AddDimension(fid, dim.ID); err != nil {
-					log.ErrorR(req, err, log.Data{"setting-response-status": http.StatusInternalServerError})
-					w.WriteHeader(http.StatusInternalServerError)
-					return
-				}
+				names = append(names, dim.ID)
 			}
+		}
 
+		fid, err := c.CreateJob(datasetModel.ID, names)
```
Comment (nshumoogum) [inline handlers/handlers.go]: CreateBlueprint?

File: handlers/handlers_test.go [hunk 2]
```diff
@@ -75,10 +73,11 @@ func TestUnitHandlers(t *testing.T) {
 
 		Convey("test CreateFilterID returns 500 if unable to create filter job on filter api", func() {
 			mockClient := NewMockFilterClient(mockCtrl)
-			mockClient.EXPECT().CreateJob(gomock.Any()).Return("", errors.New("no filter job for you"))
+			mockClient.EXPECT().CreateJob(gomock.Any(), gomock.Any()).Return("", errors.New("no filter job for you"))
```
Comment (nshumoogum) [inline handlers/handlers_test.go]: There shouldn't be any reference to filter job, there is only filter blueprints and filter outputs.

File: handlers/mock_handlers.go [hunk 3]
```diff
@@ -155,16 +155,16 @@ func (_mr *MockFilterClientMockRecorder) AddDimensionValue(arg0, arg1, arg2 inte
 }
 
 // CreateJob mocks base method
-func (_m *MockFilterClient) CreateJob(_param0 string) (string, error) {
-	ret := _m.ctrl.Call(_m, "CreateJob", _param0)
+func (_m *MockFilterClient) CreateJob(_param0 string, _param1 []string) (string, error) {
+	ret := _m.ctrl.Call(_m, "CreateJob", _param0, _param1)
```
Comment (nshumoogum) [inline handlers/mock_handlers.go]: CreateBlueprint?

File: handlers/mock_handlers.go [hunk 4]
```diff
@@ -155,16 +155,16 @@ func (_mr *MockFilterClientMockRecorder) AddDimensionValue(arg0, arg1, arg2 inte
...
 	ret0, _ := ret[0].(string)
 	ret1, _ := ret[1].(error)
 	return ret0, ret1
 }
 
 // CreateJob indicates an expected call of CreateJob
-func (_mr *MockFilterClientMockRecorder) CreateJob(arg0 interface{}) *gomock.Call {
-	return _mr.mock.ctrl.RecordCall(_mr.mock, "CreateJob", arg0)
+func (_mr *MockFilterClientMockRecorder) CreateJob(arg0, arg1 interface{}) *gomock.Call {
```
Comment (nshumoogum) [inline handlers/mock_handlers.go]: CreateBlueprint?

File: handlers/handlers_test.go [hunk 5]
```diff
@@ -75,10 +73,11 @@ func TestUnitHandlers(t *testing.T) {
 
 		Convey("test CreateFilterID returns 500 if unable to create filter job on filter api", func() {
 			mockClient := NewMockFilterClient(mockCtrl)
-			mockClient.EXPECT().CreateJob(gomock.Any()).Return("", errors.New("no filter job for you"))
+			mockClient.EXPECT().CreateBlueprint(gomock.Any(), gomock.Any()).Return("", errors.New("no filter job for you"))
```
Comment (nshumoogum) [inline handlers/handlers_test.go]: Error messages should also be updated: `errors.New("unable to create filter blueprint")`

Not sure why you have `for you` at the end of the message

Comment (mattrout92) [reply handlers/handlers_test.go]: Because it's a test and the error message never gets checked, but I will change it just for you ;)

**Pull request merged**